utopia-api/src/property-controls/property-controls.ts

```diff
@@ -42,12 +42,16 @@ export interface ColorControlDescription extends AbstractBaseControlDescription<
 }
 
 export type AllowedEnumType = string | boolean | number | undefined | null
+export interface BasicControlOption<T> {
+  value: T
+  label: string
+}
+
+export type BasicControlOptions<T> = AllowedEnumType[] | BasicControlOption<T>[]
 
 export interface PopUpListControlDescription extends AbstractBaseControlDescription<'popuplist'> {
-  defaultValue?: AllowedEnumType
-  options: AllowedEnumType[]
-  optionTitles?: string[] | ((props: unknown | null) => string[])
-  displaySegmentedControl?: boolean
+  defaultValue?: unknown
```

Comment:
why not a `T` here?

Reply:
tbh I didn't really think it would be worth it. I think using the `T` above is possibly a bit over-engineered since I don't think there would be any way for it to provide any actual use in practice

Reply:
this is the user-facing type definition, and for that it makes sense to make it a T.

when parsing it doesn't matter at all